x-pack/legacy/plugins/monitoring/public/components/cluster/overview/logstash_panel.js

```diff
@@ -45,32 +45,44 @@ export function LogstashPanel(props) {
   const setupModeLogstashData = get(setupMode.data, 'logstash');
   let setupModeInstancesData = null;
   if (setupMode.enabled && setupMode.data) {
-    const migratedNodesCount = Object.values(setupModeLogstashData.byUuid).filter(node => node.isFullyMigrated).length;
-    let totalNodesCount = Object.values(setupModeLogstashData.byUuid).length;
-    if (totalNodesCount === 0 && get(setupMode.data, 'logstash.detected.mightExist', false)) {
-      totalNodesCount = 1;
-    }
+    const {
+      totalUniqueInstanceCount,
+      totalUniqueFullyMigratedCount,
+      totalUniquePartiallyMigratedCount
+    } = setupModeLogstashData;
+    const hasInstances = totalUniqueInstanceCount > 0 || get(setupModeLogstashData, 'detected.mightExist', false);
+    const allMonitoredByMetricbeat = totalUniqueInstanceCount > 0 &&
+      (totalUniqueFullyMigratedCount === totalUniqueInstanceCount || totalUniquePartiallyMigratedCount === totalUniqueInstanceCount);
+    const internalCollectionOn = totalUniquePartiallyMigratedCount > 0;
+    if (hasInstances && (!allMonitoredByMetricbeat || internalCollectionOn)) {
+      let tooltipText = null;
 
-    const badgeColor = migratedNodesCount === totalNodesCount
-      ? 'secondary'
-      : 'danger';
+      if (!allMonitoredByMetricbeat) {
+        tooltipText = i18n.translate('xpack.monitoring.cluster.overview.logstashPanel.setupModeNodesTooltip.oneInternal', {
+          defaultMessage: `There's at least one node that isn't being monitored using Metricbeat. Click the flag
```

Comment:
Noticed there's a bit of inconsistency in terminology here: "...at least one _node_..."

APM uses "server", Beats uses "instance", and here Logstash is using "node". Is this intended?

Reply:
I think those are the correct terms, but we'll do a copy pass with gail in the final PR